src/NSubstitute/Compatibility/CompatArgInstance.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Linq.Expressions;
+
+namespace NSubstitute.Compatibility
+{
+    /// <summary>
+    /// Alternate version of <see cref="Arg"/> matchers for compatibility with pre-C#7 compilers
+    /// which do not support <c>ref</c> return types. Do not use unless you are unable to use <see cref="Arg"/>.
+    /// 
+    /// <c>CompatArgInstance</c> provides a non-static version of <see cref="CompatArg"/> which can make it easier
+    /// to use from an abstract base class.
+    /// </summary>
```

Comment:
Should we probably add a sample of usage here?

Reply:
Replacing with link to docs as discussed in https://github.com/nsubstitute/NSubstitute/pull/475#discussion_r230937534